vignettes/tool-calling.Rmd

```diff
@@ -74,34 +76,36 @@ get_current_time <- function(tz = "UTC") {
 }
 ```
 
-Note that we've gone through the trouble of creating [roxygen2 comments](https://roxygen2.r-lib.org/). This is a very important step that will help the model use your tool correctly!
-
-Let's test it:
+To turn a function into a tool, we provide some additional metadata that the model will use:
 
 ```{r}
-get_current_time()
+get_current_time <- tool(
+  get_current_time,
+  description = "Gets the current time in the given time zone.",
+  arguments = list(
+    tz = type_string(
+      "The time zone to get the current time in. Defaults to `\"UTC\"`.",
+      required = FALSE
+    )
+  )
+)
 ```
 
-### Registering tools
+This is a fair amount of code to write, even for such a simple function. Fortunately, you don't have to write this by hand! I generated the above `tool()` call by calling `create_tool_def(get_current_time)`, which printed that code at the console. `create_tool_def()` works by passing the function's signature and documentation to an LLM and asking it to generate the `tool()` call for you. `create_tool_def()` is not perfect, so you must review the generated code before using it. But it is a huge time-saver nonetheless, and removes the boilerplate generation you'd have to do otherwise.
 
-Now we need to tell our chat object about our `get_current_time` function. This by creating and registering a tool:
+Note that a tool is just a special type of function so we can still call it:
 
 ```{r}
-chat$register_tool(tool(
-  get_current_time,
-  "Gets the current time in the given time zone.",
-  tz = type_string(
-    "The time zone to get the current time in. Defaults to `\"UTC\"`.",
-    required = FALSE
-  )
-))
+get_current_time()
 ```
 
-This is a fair amount of code to write, even for such a simple function as `get_current_time`. Fortunately, you don't have to write this by hand! I generated the above `register_tool` call by calling `create_tool_def(get_current_time)`, which printed that code at the console. `create_tool_def()` works by passing the function's signature and documentation to GPT-4o, and asking it to generate the `register_tool` call for you.
+### Registering and using tools
 
-Note that `create_tool_def()` may not create perfect results, so you must review the generated code before using it. But it is a huge time-saver nonetheless, and removes the tedious boilerplate generation you'd have to do otherwise.
+Now we need to tell our chat object about our tool. We do this with `$register_tool()`:
```

Comment:
```suggestion
Now we need to give our chat object access to our tool. We do this with `$register_tool()`:
```

I normally like "tell X about Y", but I think it's better to think of `description` as how you tell your chat object about the tool -- or really _about how to use_ the tool, but _about_ is ambiguous as currently written.